### DISCUSSION TITLE
Popup's firewall filter widget

### DISCUSSION THREAD

**Discussion by u-RraaLL** [uBlockOrigin/uBlock-issues#2632] [Q&A] [answered]
Is the new firewall filter widget in uBO's popup supposed the remember the settings across websites? And even after closing the browser?

**Comment by ghost** [comment]
Yes, saved for me.

https://github.com/gorhill/uBlock/commit/8d445e782ddc57158e0bffcc08dec193340d49a0#diff-39353dbeaba500b72e753acb6dba44fe833b32c748b558a8db7780e945f07833R509-R513

But I would like to see some better "tip" that rows are hidden - bold borders between rows, where hidden row should be or maybe some arrow pointing down at the bottom, maybe clickable, to show all rows.

**Reply by gorhill** [reply]
How about this?

![Screenshot from 2023-07-21 09-38-42](https://github.com/uBlockOrigin/uBlock-issues/assets/585534/6da80331-f262-4210-945b-212f4b503135)

**Reply by ghost** [reply]
Good enough.

**Reply by gothic-bum** [reply]
Good solution, but I noticed that portions of the text for collapsed CNAME rows are being shown:

<img width="381" alt="cname_row_artifacts" src="https://github.com/uBlockOrigin/uBlock-issues/assets/76543584/acce38b2-3950-44a9-8884-0e717be3c711">

**Reply by gorhill** [reply, marked as answer]
https://github.com/gorhill/uBlock/commit/fdb6dc2cc231128752eb25cb94ce87ce7b052b7b

**Comment by u-RraaLL** [comment]
> Yes, saved for me.

I was actually more curious about the reasoning behind the settings getting saved rather than if it affected only myself.

**Reply by gorhill** [reply]
The idea is that some people might want to _only_ see what's blocked, since it's these rows that are of interest when trying to unbreak a site.

**Reply by u-RraaLL** [reply]
Could you perhaps let us make a choice there to make the filtering subset permanent or not? E.g. by clicking a lock icon in the filtering pane itself, similar to per-site -switches or dynamic filtering.

If not, the visual cue you suggested gwarser could be enough by itself.

**Comment by I-I-IT** [comment]
Except this thread, this is not in the Wiki or the documentation. I believe it should be added. Something like "Collapsed rows are normal and show requests blocked by the network firewall directly" (correct me if wrong).

**Reply by garry-ut99** [reply]
Updated.

**Reply by u-RraaLL** [reply]
That's not a dynamic filtering feature though. It should be put in the quick guide for the popup user interface instead: https://github.com/gorhill/uBlock/wiki/Quick-guide:-popup-user-interface

**Reply by garry-ut99** [reply]
Ok, moved and improved: [LINK](https://github.com/uBlockOrigin/uBlock-issues/wiki/Quick-guide:-popup-user-interface#the-overview-panel). 
By the way, I was also thinking about combining 2 "forbes.com" gifs into one.
